spec/repository_spec.rb

```diff
@@ -1,37 +1,66 @@
 describe 'datadog::repository' do
+  context 'on debianoids' do
+    cached(:chef_run) do
+      ChefSpec::SoloRunner.new(
+        platform: 'debian', version: '8.5'
+      ).converge(described_recipe)
+    end
+
+    it 'include apt cookbook' do
+      expect(chef_run).to include_recipe('apt::default')
+    end
+
+    it 'installs apt-transport-https' do
+      expect(chef_run).to install_package('apt-transport-https')
+    end
+
+    it 'installs new apt key' do
+      expect(chef_run).to run_execute('apt-key import key 382E94DE').with(
+        command: 'apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 A2923DFF56EDA6E76E55E492D3A80E30382E94DE'
+      )
+    end
+
+    it 'sets up an apt repo' do
+      expect(chef_run).to add_apt_repository('datadog')
+    end
+  end
+
   context 'rhellions' do
-    describe 'on versions 5.x and lower' do
-      cached(:chef_run) do
-        ChefSpec::SoloRunner.new(
-          :platform => 'centos',
-          :version => '5.8'
-        ) do |node|
-          node.set['languages'] = { 'python' => { 'version' => '2.4.3' } }
-        end.converge described_recipe
-      end
-
-      it 'sets up a yum repo' do
-        expect(chef_run).to create_yum_repository('datadog').with(
-          gpgkey: 'http://yum.datadoghq.com/DATADOG_RPM_KEY.public'
-        )
-      end
```

Comment:
Actually there's one RHEL5-specific thing: the yum repository has to be configured with a gpgkey downloaded through plain text `http` (the logic that handles that is in the default attributes file: https://github.com/DataDog/chef-datadog/blob/v2.8.1/attributes/default.rb#L85-L91).